partiql-eval/src/main/kotlin/org/partiql/eval/internal/operator/rel/RelSort.kt

```diff
@@ -0,0 +1,67 @@
+package org.partiql.eval.internal.operator.rel
+
+import org.partiql.eval.internal.Record
+import org.partiql.eval.internal.operator.Operator
+import org.partiql.plan.Rel
+import org.partiql.value.NullOrder
+import org.partiql.value.PartiQLValueComparator
+import org.partiql.value.PartiQLValueExperimental
+
+@OptIn(PartiQLValueExperimental::class)
+internal class RelSort(
+    val input: Operator.Relation,
+    val specs: List<Pair<Operator.Expr, Rel.Op.Sort.Order>>
+
+) : Operator.Relation {
+    private var records: MutableList<Record> = mutableListOf()
```

Comment:
Changed to have `records` be an `Iterator<Record>`, which allows us to use `records.next()` rather than `removeAt`. I think `records` directly as an `Iterator` may be cleaner than having `records` be an `Iterable` with a backing collection, but let me know if there's some simpler representation.